### ISSUE TITLE
Update MaStR data

### ISSUE THREAD

**Issue by nesnoj** [openego/eGon-data#519]
There will be a new release of the MaStR unit registry soon..

**Comment by nesnoj** [comment]
@ClaraBuettner I just received the new version and will integrate into the bundle in February..

**Comment by nesnoj** [comment]
Will not integrate into data bundle, will make a new release on Zenodo instead.

**Comment by nesnoj** [comment]
Also check if #134 is still relevant

**Comment by nesnoj** [comment]
@KathiEsterl @CarlosEpia some notes for your experiments:
- You can use this issue and branch [`features/#519-update-mastr-data`](https://github.com/openego/eGon-data/tree/features/%23519-update-mastr-data) (I already made some adjustments there, cf. ) for your work. In parallel, I will code the import of the data into DB tables in #1051 to quickly get a working dataset even if you decide not to use the new data.
- Data is located on your server here: `/home/egondata/mastr/data/` .
- I propose to use units before 2022-01-01 (column `Inbetriebnahmedatum`), is this ok @IlkaCu @ClaraBuettner 
- The new dataset is much more consistent (some cell values are distinct, units have been approved by the grid operator) compared to the old one.
- But some stuff has changed in the new dataset: column names, values. Also, data in the new dump is split into more files (_example:_ before: files with units, 1 file for locations, new: files with units, 1 file for locations, 1 file with connection points)
  - I **preprocessed and joined** the raw data to with the aim to change the existing code as little as possible using the same file names. In doing so, I was able to make many tasks work.
  - When you start a SH run it will crash at `wind_farms.py` which I was not able to fix quickly (see below).
  - So in the best case you should just be able to work with the new files out of the box, let me know if further adjustments are necessary.

_______________________________
Note on `wind_farms.py`:
[Here](https://github.com/openego/eGon-data/blob/f18de38166ae42709481c138706df8c8197c8138/src/egon/data/datasets/power_plants/wind_farms.py#L167-L174) the merge of WT and locations take place but it does not work due to the new structure. I experimented with the following snippet (unfinished):
```python
wea = wea.merge(
    bus[["MaStRNummer", "Spannungsebene"]],
    left_on="LokationMastrNummer",
    right_on="MaStRNummer",
    how="left"
)
wea.rename(columns={"Spannungsebene": "connection point"}, inplace=True)
wea.rename(columns={"Spannungsebene": "voltage"}, inplace=True)
```

**Comment by nesnoj** [comment]
BTW, this seems to be a typo (should be level 2, not 1)?
https://github.com/openego/eGon-data/blob/4a2d0168ccd3deca5d8333ba1442b5b47ada90ab/src/egon/data/datasets/power_plants/pv_ground_mounted.py#L95-L96

**Comment by nesnoj** [comment]
Here's how the new dataset is imported into the DB:
https://github.com/openego/eGon-data/pull/1054/files#diff-493e771cad7898add452db376099be7efc77aed013de0b728ac5da2e087c1b3d

Just for your information. You already have an import, I just want to keep you up-to-date..

**Comment by KathiEsterl** [comment]
> 

We do not have a voltage level 2, so this is right: 
https://redmine.iks.cs.ovgu.de/oe/projects/ego-n/wiki/Definition_of_thresholds_for_voltage_level_assignment
You can either keep it as it is in the MaStR-import-function or also adapt it, I gues...

Thank you for all the information! Carlos and I will have a deeper look into it today.

**Comment by CarlosEpia** [comment]
Hi @nesnoj 
We tried to execute egon-data in test mode and the task power_plants.mastr.import-mastr failed with this error message:
...
[2022-11-29 14:23:21,930] {logging_mixin.py:120} INFO -   Writing to DB...
[2022-11-29 14:23:22,400] {taskinstance.py:1150} ERROR - value too long for type character varying(11)
CONTEXT:  COPY egon_power_plants_pv, line 505, column orientation_primary: "Nachgefuehrt"
Traceback (most recent call last):
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/home/student/egon-data/eGon-data/src/egon/data/datasets/__init__.py", line 194, in skip_task
    result = super(type(task), task).execute(*xs, **ks)
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 113, in execute
    return_value = self.execute_callable()
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/student/egon-data/eGon-data/src/egon/data/datasets/power_plants/mastr.py", line 325, in import_mastr
    units.to_postgis(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/geopandas/geodataframe.py", line 1808, in to_postgis
    geopandas.io.sql._write_postgis(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/geopandas/io/sql.py", line 431, in _write_postgis
    gdf.to_sql(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/pandas/core/generic.py", line 2872, in to_sql
    sql.to_sql(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/pandas/io/sql.py", line 717, in to_sql
    pandas_sql.to_sql(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/pandas/io/sql.py", line 1761, in to_sql
    sql_engine.insert_records(
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/pandas/io/sql.py", line 1340, in insert_records
    table.insert(chunksize=chunksize, method=method)
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/pandas/io/sql.py", line 967, in insert
    exec_insert(conn, keys, chunk_iter)
  File "/home/student/egon-data/venv/lib/python3.8/site-packages/geopandas/io/sql.py", line 312, in _psql_insert_copy
    cur.copy_expert(sql=sql, file=s_buf)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(11)
CONTEXT:  COPY egon_power_plants_pv, line 505, column orientation_primary: "Nachgefuehrt"

[2022-11-29 14:23:22,418] {taskinstance.py:1187} INFO - Marking task as FAILED. dag_id=egon-data-processing-pipeline, task_id=power_plants.mastr.import-mastr, execution_date=20221129T110706, start_date=20221129T132240, end_date=20221129T132322
[2022-11-29 14:23:25,819] {local_task_job.py:102} INFO - Task exited with return code 1

Maybe we can try to find a solution tomorrow during the AP1 meeting.

**Comment by KathiEsterl** [comment]
Hi Jonathan, 

sorry, we did'nt understand your former explanation correct, but now everything is working!

It really looks like there is nearly nothing to change in our scripts which is really nice! 
We will have a look on the results now and then push our changes to this branch: `features/#519-update-mastr-data`

We will let you know when we are finished.

Thanks for your work.

**Comment by nesnoj** [comment]
We agreed on using the new dataset only for pv, wind & hydro. Hence, we will use 2 different datasets.
For testing, please follow the workflow [above](https://github.com/openego/eGon-data/issues/519#issuecomment-1331000299) and work in branch `features/#519-update-mastr-data`.

Next steps - please tick when you are done:
- [ ] Adapt and check ground-mounted PV @KathiEsterl 
- [ ] Adapt and check wind @CarlosEpia 
- [ ] Adapt and check hydro @IlkaCu 
- [ ] Create additional zenodo repo for new dataset and integrate download @nesnoj 
- [ ] Adapt distribution of rooftop PV to buildings @nesnoj 
- [ ] Merge creation of SQ tables #1051 into this branch @nesnoj 

If we're quick, this can be part of the upcoming CI run #1057.

Thank you!

**Comment by nesnoj** [comment]
> [ ] Adapt and check hydro @IlkaCu

@IlkaCu I just discovered that the dataset contains only integers in `ArtDerWasserkraftanlage` (which specifies run-of-river, ...). I have to fix the dataset accordingly before you start adapting the hydro part.

**Comment by nesnoj** [comment]
> > [ ] Adapt and check hydro @IlkaCu
> 
> @IlkaCu I just discovered that the dataset contains only integers in `ArtDerWasserkraftanlage` (which specifies run-of-river, ...), cf. [OpenEnergyPlatform/open-MaStR#390](https://github.com/OpenEnergyPlatform/open-MaStR/issues/390). I have to fix the dataset accordingly before you start adapting the hydro part.

Fixed! I replaced `bnetza_mastr_hydro_cleaned.csv` on your server.
FYI: The dump is also available on [zenodo](https://sandbox.zenodo.org/record/1132987) now.

**Comment by KathiEsterl** [comment]
@nesnoj : Carlos pushed our change to `features/#519-update-mastr-data` :+1:

**Comment by nesnoj** [comment]
Thank you. I'll test tonight if it plays well with my latest changes in #1054 for SH. If so, I'll merge into CI and start the run tomorrow #1057.
